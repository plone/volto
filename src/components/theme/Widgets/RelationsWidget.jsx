import React from 'react';
import cx from 'classnames';
import RelationWidget from './RelationWidget';

const RelationsWidget = ({ value, children, className }) =>
  value ? (
    <ul className={cx(className, 'relations', 'widget')}>
      {value.map((item, key) => (
        <li key={key}>
          <RelationWidget
            value={item}
            className={className}
            key={item.token || item.title || item}
          >
            {children}
          </RelationWidget>
        </li>
      ))}
    </ul>
  ) : (
    ''
  );

export default RelationsWidget;
