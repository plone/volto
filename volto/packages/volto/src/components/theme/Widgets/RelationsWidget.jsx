import React from 'react';
import cx from 'classnames';
import RelationWidget from './RelationWidget';

const RelationsWidget = ({ value, children, className }) =>
  value ? (
    <ul className={cx(className, 'relations', 'widget')}>
      {value.map((item, key) => {
        return item ? (
          <li key={key}>
            <RelationWidget
              value={item || `relation target not found '${key}'`}
              className={className}
              key={item.token || item.title || item}
            >
              {children}
            </RelationWidget>
          </li>
        ) : null;
      })}
    </ul>
  ) : (
    ''
  );

export default RelationsWidget;
