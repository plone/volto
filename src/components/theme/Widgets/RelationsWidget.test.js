import React from 'react';
import renderer from 'react-test-renderer';
import RelationsWidget from './RelationsWidget';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-intl-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const mockStore = configureStore([thunk]);

const store = mockStore({
  intl: {
    locale: 'en',
    messages: {},
  },
  userSession: {
    token: '12345',
  },
});

describe('RelationsWidget', () => {
  it('renders an empty relations view widget component', () => {
    const component = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <RelationsWidget />
        </Provider>
      </MemoryRouter>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders a simple relations view widget component', () => {
    const component = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <RelationsWidget className="metadata" value={['foo', 'bar']} />
        </Provider>
      </MemoryRouter>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders a vocabulary relations view widget component', () => {
    const component = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <RelationsWidget
            className="metadata"
            value={[{ title: 'Foo' }, { title: 'Bar' }]}
          />
        </Provider>
      </MemoryRouter>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders a full vocabulary array view widget component', () => {
    const component = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <RelationsWidget
            className="metadata"
            value={[
              { title: 'Foo', token: 'foo' },
              { title: 'Bar', token: 'bar' },
            ]}
          />
        </Provider>
      </MemoryRouter>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders a full vocabulary relations view widget component with children', () => {
    const component = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <RelationsWidget
            className="metadata"
            value={[
              {
                title: 'Foo',
                token: 'foo',
                '@id': '/foo-page',
                '@type': 'Foo',
                review_state: 'private',
                description: 'A Foo',
              },
              {
                title: 'Bar',
                token: 'bar',
                '@id': '/bar-page',
                '@type': 'Bar',
                review_state: 'published',
                description: 'A Bar',
              },
            ]}
          >
            {(child) => <strong>{child}</strong>}
          </RelationsWidget>
        </Provider>
      </MemoryRouter>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
