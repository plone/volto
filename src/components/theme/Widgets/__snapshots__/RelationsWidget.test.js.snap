// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RelationsWidget renders a full vocabulary array view widget component 1`] = `
<ul
  className="metadata relations widget"
>
  <li>
    <a
      className="metadata relation widget"
      href="/"
      onClick={[Function]}
      target={null}
      title=""
    >
      <svg
        className="icon icon-margin"
        dangerouslySetInnerHTML={
          Object {
            "__html": undefined,
          }
        }
        onClick={null}
        style={
          Object {
            "fill": "#878f93",
            "height": "20px",
            "width": "auto",
          }
        }
        viewBox=""
        xmlns=""
      />
      Foo
    </a>
  </li>
  <li>
    <a
      className="metadata relation widget"
      href="/"
      onClick={[Function]}
      target={null}
      title=""
    >
      <svg
        className="icon icon-margin"
        dangerouslySetInnerHTML={
          Object {
            "__html": undefined,
          }
        }
        onClick={null}
        style={
          Object {
            "fill": "#878f93",
            "height": "20px",
            "width": "auto",
          }
        }
        viewBox=""
        xmlns=""
      />
      Bar
    </a>
  </li>
</ul>
`;

exports[`RelationsWidget renders a full vocabulary relations view widget component with children 1`] = `
<ul
  className="metadata relations widget"
>
  <li>
    <a
      className="metadata relation widget Foo private"
      href="/foo-page"
      onClick={[Function]}
      target={null}
      title="A Foo"
    >
      <svg
        className="icon icon-margin"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<title>Foo</title>undefined",
          }
        }
        onClick={null}
        style={
          Object {
            "fill": "#878f93",
            "height": "20px",
            "width": "auto",
          }
        }
        viewBox=""
        xmlns=""
      />
      <strong>
        Foo
      </strong>
    </a>
  </li>
  <li>
    <a
      className="metadata relation widget Bar published"
      href="/bar-page"
      onClick={[Function]}
      target={null}
      title="A Bar"
    >
      <svg
        className="icon icon-margin"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<title>Bar</title>undefined",
          }
        }
        onClick={null}
        style={
          Object {
            "fill": "#878f93",
            "height": "20px",
            "width": "auto",
          }
        }
        viewBox=""
        xmlns=""
      />
      <strong>
        Bar
      </strong>
    </a>
  </li>
</ul>
`;

exports[`RelationsWidget renders a simple relations view widget component 1`] = `
<ul
  className="metadata relations widget"
>
  <li>
    <a
      className="metadata relation widget"
      href="/"
      onClick={[Function]}
      target={null}
      title=""
    >
      <svg
        className="icon icon-margin"
        dangerouslySetInnerHTML={
          Object {
            "__html": undefined,
          }
        }
        onClick={null}
        style={
          Object {
            "fill": "#878f93",
            "height": "20px",
            "width": "auto",
          }
        }
        viewBox=""
        xmlns=""
      />
      foo
    </a>
  </li>
  <li>
    <a
      className="metadata relation widget"
      href="/"
      onClick={[Function]}
      target={null}
      title=""
    >
      <svg
        className="icon icon-margin"
        dangerouslySetInnerHTML={
          Object {
            "__html": undefined,
          }
        }
        onClick={null}
        style={
          Object {
            "fill": "#878f93",
            "height": "20px",
            "width": "auto",
          }
        }
        viewBox=""
        xmlns=""
      />
      bar
    </a>
  </li>
</ul>
`;

exports[`RelationsWidget renders a vocabulary relations view widget component 1`] = `
<ul
  className="metadata relations widget"
>
  <li>
    <a
      className="metadata relation widget"
      href="/"
      onClick={[Function]}
      target={null}
      title=""
    >
      <svg
        className="icon icon-margin"
        dangerouslySetInnerHTML={
          Object {
            "__html": undefined,
          }
        }
        onClick={null}
        style={
          Object {
            "fill": "#878f93",
            "height": "20px",
            "width": "auto",
          }
        }
        viewBox=""
        xmlns=""
      />
      Foo
    </a>
  </li>
  <li>
    <a
      className="metadata relation widget"
      href="/"
      onClick={[Function]}
      target={null}
      title=""
    >
      <svg
        className="icon icon-margin"
        dangerouslySetInnerHTML={
          Object {
            "__html": undefined,
          }
        }
        onClick={null}
        style={
          Object {
            "fill": "#878f93",
            "height": "20px",
            "width": "auto",
          }
        }
        viewBox=""
        xmlns=""
      />
      Bar
    </a>
  </li>
</ul>
`;

exports[`RelationsWidget renders an empty relations view widget component 1`] = `""`;
